src/rard/research/models/text_object_field.py

```diff
@@ -22,8 +22,10 @@ def related_lock_object(self):
     references = GenericRelation("BibliographyItem", related_query_name="text_fields")
 
     def get_history_title(self):
+        from rard.research.models import Book, Work
+
```

Comment:
Can get class name as string here to save importing models:
```suggestion
        from rard.research.models import Book, Work
        obj = self.get_related_object()
        if obj.__class__.__name__ in ["Antiquarian", "Work", "Book"]:
```

Reply:
Annoying, suggestion doesn't work properly because it's across a deleted line, but you get the gist.